src/dune_rules/context.ml

```diff
@@ -302,20 +297,36 @@ module Build_environment_kind = struct
           match opam_prefix with
           | Some s -> Opam2_environment s
           | None -> Unknown)))
-end
 
-let ocamlfind_printconf_path ~env ~ocamlfind ~toolchain =
-  let args =
-    let args = [ "printconf"; "path" ] in
-    match toolchain with
-    | None -> args
-    | Some s -> "-toolchain" :: Context_name.to_string s :: args
-  in
-  let+ l =
-    Memo.of_reproducible_fiber
-      (Process.run_capture_lines ~display:Quiet ~env Strict ocamlfind args)
-  in
-  List.map l ~f:Path.of_filename_relative_to_initial_cwd
+  let findlib_paths findlib ~kind ~findlib_toolchain ~env ~dir =
+    match (findlib, query ~kind ~findlib_toolchain ~env) with
```

Comment:
This is a bit of a matter of taste, but I'm not a function that takes a bunch of arguments only to apply them to another function. Why not pass the result of the function directly then? E.g.

```
let findlib_paths t findlib ~dir =
  match (findlib, t) with ...
```